### ISSUE TITLE
Multiple Binaries / Path to source

### ISSUE THREAD

**Issue by zet4** [goreleaser/goreleaser#49]
So creating another issue based on my use-cases of goreleaser,
Ran into issue of getting goreleaser to work with [whats-this/owo.go](https://github.com/whats-this/owo.go) where I have `owo.go/cmd/gowo`, CLI application in a nested folder, goreleaser doesn't see it from root and fails to run because root directory which itself is the library doesn't have a main.go.

Currently to release gowo I had to copy license, create separate readme.md (would be nice to preserve this but not necessary) and move yml config for releaser to `cmd/gowo`. 
But there would issue if I was to add additional tools aside from gowo (another folder in `cmd` for ex) as the archive would get overwritten (if at all committed, releaser errors if release already exists)

Basically what I would like to request is for either goreleaser to build all things `./...` instead of just one
Or make it easier to add non-root `main.go` locations.

Sorry for giving so many requests but I really like this project and would love to see it support more use-cases

**Comment by caarlos0** [comment]
Hi @zet4,

This is already supported.

See https://github.com/goreleaser/releaser#build-customization:

```yaml
build:
  main: ./cmd/main.go
```

**Comment by zet4** [comment]
Not entirely though, if I had multiple tools inside a repo I would still have this issue.

**Comment by caarlos0** [comment]
Oh, indeed, this is not supported yet...

**Comment by caarlos0** [comment]
@zet4 I was thinking about it: is this a good practice? Do you know any more projects that do this?

**Comment by jorinvo** [comment]
One way would be to allow passing a list to `main`:

``` yml
build:
  main: 
    - ./cmd/first/main.go
    - ./cmd/second/main.go
```

I also think this is a use case for many projects. 
For tools it might be less common but for products you often have more than one binary in a repository.

**Comment by caarlos0** [comment]
@jorinvo this would also break the `binary_name` option...

I'll have to think better about this...

**Comment by jorinvo** [comment]
@caarlos0 good point!
Still would be a nice feature in the future  :)

**Comment by jorinvo** [comment]
I think we should include this in v1.0.

I would suggest to add a `builds` key.
If you like to have multiple binaries you could. specify `builds` instead of `build`.
`builds` would be a list of objects containing the same options as `build`.
This way you can change all options for each binary independently.
Probably also good to only allow one at a time - `build` OR `builds`. 

``` yml
build:
  main: ./cmd/binone/main.go
  ldflags: -s -w
  goos:
    - freebsd
    - windows
```

OR

``` yml
builds:
  - main: ./cmd/binone/main.go
    ldflags: -s -w
    goos:
      - freebsd
      - windows
  - main: ./cmd/bintwo/main.go
    ldflags: -s -w
    goos:
      - linux
      - darwin
```

What do you think? Has someone a prettier solution?

**Comment by caarlos0** [comment]
@jorinvo yeah, that makes sense... in that case we would also have two brew formulas, right?

**Comment by caarlos0** [comment]
Just realized that this is now easy to do since we now have `binary_name` inside the `build` section 👍

**Comment by jorinvo** [comment]
@caarlos0 but you're right that then you still couldn't define brew `caveats` and dependencies separately..

**Comment by caarlos0** [comment]
yeah, there is that..

maybe we can do something different, like:

```yaml
foo:
  build:
  brew:
  etc:
bar:
  build:
  brew:
  etc:
```

what do you think?

**Comment by caarlos0** [comment]
or maybe it's just easier to define multiple gorelaser.yml files and call goreleaser two times:

```sh
goreleaser -c foo.goreleaser.yml
goreleaser -c bar.goreleaser.yml
```

**Comment by caarlos0** [comment]
Works like a charm: https://github.com/caarlos0/test/releases/tag/v1.0.1

**Comment by caarlos0** [comment]
hey @zet4 , does having multiple `goreleaser.yml` files solves the issue for you?

If yes, IMO, we can close this :)

**Comment by zet4** [comment]
@caarlos0 yeah that could work, I guess I will close this then

**Comment by jorinvo** [comment]
@caarlos0, I didn't consider this an option because I thought people still want to bundle the different binaries in the same archive file.
We can still think about another solution as soon as someone actually needs it :)

**Comment by caarlos0** [comment]
@jorinvo Yeah, if the objective is to build multiple binaries and release them both in the same package and brew formula, then the previous noted solution would work...

**Comment by caarlos0** [comment]
I'll implement that so we can test how it works :)

**Comment by hellais** [comment]
This is currently the limiting fact for me using goreleaser for my releases. I have a single package where I would like to ship multiple binaries that each are generated from a separate `main.go`. I would like all the binaries to end up inside of the final `.tar.gz` so having separate `goreleaser.yml` files is not an option.

Could this issue be re-opened or should I file another one?

**Comment by martinhoefling** [comment]
@hellais +1

**Comment by caarlos0** [comment]
@hellais @martinhoefling in this case I think it makes sense... I'll create a new issue for that case, since this one was slightly different.

**Comment by caarlos0** [comment]
check #259

**Comment by caarlos0** [comment]
released on v0.22.0
